playbooks/common/openshift-cluster/upgrades/v3_9/upgrade_control_plane.yml

```diff
@@ -52,48 +39,26 @@
     l_upgrade_docker_target_hosts: "oo_masters_to_config:oo_etcd_to_config"
     l_upgrade_excluder_hosts: "oo_masters_to_config"
     openshift_protect_installed_version: False
-  when: hostvars[groups.oo_first_master.0].openshift_currently_installed_version | version_compare('3.8','<')
 
-- name: Flag pre-upgrade checks complete for hosts without errors 3.8
-  hosts: oo_masters_to_config:oo_etcd_to_config
-  tasks:
-  - set_fact:
-      pre_upgrade_complete: True
-    when: hostvars[groups.oo_first_master.0].openshift_currently_installed_version | version_compare('3.8','<')
-
-# Pre-upgrade completed
-
-- name: Intermediate 3.8 Upgrade
-  import_playbook: ../upgrade_control_plane.yml
-  when: hostvars[groups.oo_first_master.0].openshift_currently_installed_version | version_compare('3.8','<')
-
-## 3.8 upgrade complete we should now be able to upgrade to 3.9
-
-- name: Configure the upgrade target for the common upgrade tasks 3.9
+## If the cluster is running 3.7, perform and intermediate upgrade to 3.8
+- name: Configure the upgrade target for the common upgrade tasks 3.8
   hosts: oo_masters_to_config:oo_etcd_to_config:oo_lb_to_config
   tasks:
-  - meta: clear_facts
   - set_fact:
-      openshift_upgrade_target: '3.9'
-      openshift_upgrade_min: '3.8'
-      openshift_release: '3.9'
-      openshift_pkg_version: "{{ _requested_pkg_version if _requested_pkg_version is defined else '' }}"
-  # Set the user's specified image_tag for 3.9 upgrade if it was provided.
-  - set_fact:
-      openshift_image_tag: "{{ _requested_image_tag }}"
-      l_force_image_tag_to_version: False
-    when: _requested_image_tag is defined
-  # If the user didn't specify an image_tag, we need to force update image_tag
-  # because it will have already been set during 3.8.  If we aren't running
-  # a double upgrade, then we can preserve image_tag because it will still
-  # be the user provided value.
-  - set_fact:
-      l_force_image_tag_to_version: True
-    when:
-    - l_double_upgrade_cp is defined and l_double_upgrade_cp
-    - _requested_image_tag is not defined
+      openshift_upgrade_target: '3.8'
+      openshift_upgrade_min: '3.7'
+      # Preserve these values for the 3.9 upgrade later
+      _requested_release: "{{ openshift_release }}"
+      _requested_pkg_version: "{{ openshift_pkg_version }}"
+      _requested_image_tag: "{{ openshift_image_tag }}"
+      # Set these values for the 3.8 upgrade
+      openshift_release: '3.8'
+      openshift_pkg_version: '-3.8'
+      openshift_image_tag: 'v3.8'
```

Comment:
We should preserve and set openshift_version here as well, then we can get rid of a lot of the logic.